Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -1,8 +1,14 @@
 #include "BatteryStateOfChargeService.h"
+#include "BatteryData.cpp"
```

Comment:
Including a source file inside another source file is a **VERY bad** thing. It can lead to `multiple definition` errors, which is actually what I got when switching between branches.

What ends up happening is C++ will try to compile `BatteryStateOfChargeService.cpp`, and since it includes `BatteryData.cpp`, then that obj file will also have symbols from `BatteryData.cpp`. Since .cpps don't have include guards, it will then try to compile `BatteryData.cpp`, and it'll be super confused since the symbols are already defined.

I think it worked for you since QT must be smart enough to realise when it's already built a symbol, but it's not perfect and will cause lots of cancer down the line if the compiler changes or a project gets bigger. 

In other projects, you might see people using c++ templates by including a `.cpp` somewhere, but that's still pretty cancerous, and they should be using `_impl.h` instead.